templates/fine-tune-function-calling/README.md

```diff
@@ -0,0 +1,478 @@
+
+
+
+    - limit=1 1:   0%|          | 0/1 [00:00<?, ?it/s]
+
+
+
+    Running 0:   0%|          | 0/1 [00:00<?, ?it/s]
+
+
+
+```python
+pprint_example(first_ex, dataset_format=DatasetFormat.GLAIVE)
+```
+
+    [91msystem: [0mSYSTEM: You are a helpful assistant with access to the following functions. Use them if required -
+    {
+        "name": "create_reminder",
+        "description": "Create a reminder for a specific date and time",
+        "parameters": {
+            "type": "object",
+            "properties": {
+                "reminder_text": {
+                    "type": "string",
+                    "description": "The content of the reminder"
+                },
+                "reminder_date": {
+                    "type": "string",
+                    "format": "date",
+                    "description": "The date of the reminder"
+                },
+                "reminder_time": {
+                    "type": "string",
+                    "format": "time",
+                    "description": "The time of the reminder"
+                }
+            },
+            "required": [
+                "reminder_text",
+                "reminder_date",
+                "reminder_time"
+            ]
+        }
+    }
+    
+    [92mchat: [0mUSER: I need to set a reminder for my doctor's appointment.
+    
+    
+    ASSISTANT: Sure, I can help with that. Could you please provide me with the date and time of your appointment? <|endoftext|>
+    
+    
+    USER: The appointment is on 2022-09-15 at 10:00 AM.
+    
+    
+    ASSISTANT: <functioncall> {"name": "create_reminder", "arguments": '{"reminder_text": "Doctor's appointment", "reminder_date": "2022-09-15", "reminder_time": "10:00"}'} <|endoftext|>
+    
+    
+    FUNCTION RESPONSE: {"status": "success", "message": "Reminder for 'Doctor's appointment' on 2022-09-15 at 10:00 AM has been created successfully."}
+    
+    
+    ASSISTANT: Your reminder for the doctor's appointment on 2022-09-15 at 10:00 AM has been created successfully. You will be notified at the specified time. <|endoftext|>
+    
+    
+    
+    
+
+
+If you notice, each sample has two entries: system and chat. This dataset is already formatted in specific way (e.g. using USER, \<|endoftext|\> and other tokens). To enable fine-tuning on various open source models we need to convert each row to a more general format like the OpenAI chat format, which is the preferred format for fine-tuning instruction-tuned models on Anyscale ([dataset format guide](https://docs.endpoints.anyscale.com/endpoints/fine-tuning/dataset-prep)). Let's first bring this dataset into the conversation format and inspect how that looks like:
+
+
+```python
+import fc_utils.preprocessing
+import importlib
+importlib.reload(fc_utils.preprocessing)
+from fc_utils.preprocessing import pprint_example
+```
```

Comment:
what are these? can be removed?

Reply:
I was still cleaning this up! Removed now with all new changes incorporated.